doc/reference/actions/needed_deps.rst

```diff
@@ -0,0 +1,20 @@
+needed_deps
+----
+
+.. highlight:: dune
+
+.. describe:: (needed_deps <file> ...)
+
+   Parse the files to extract defined dependencies, and declare them as
+   needed_deps for the rule. In other words, whenever any of the dependencies
+   declared as needed_deps change, the rule is rerun. 
+   The file passed as an argument contains an S-epression constructed from the
+   following atoms:
+   ``file <filename>
+    alias <alias_name>
+    file_selector <glob>
+    universe``
```

Comment:
Here we need a fuller description of each constructor. See for example the section on "Dependency Specification" in the Dune manual. https://dune.readthedocs.io/en/stable/concepts/dependency-spec.html